examples/jenkins/pipeline/pipelinetemplate.json

```diff
@@ -19,6 +19,9 @@
         "creationTimestamp": null,
         "labels": {
           "name": "sample-pipeline"
+        },
+        "annotations": {
+          "downstream.openshift.io/deployments": "[{\"name\": \"frontend\", \"namespace\": \"\"}]"
```

Comment:
The immediate need is to show the pipeline on the overview in the right place, which is why we need the BC -> DC annotation. We'd need to think about how we'd show other relationships, although even just having links between things is helpful. Maybe on the detailed pipeline page.

@jwforres